modules/common/tls/tls.go

```diff
@@ -283,12 +282,15 @@ func (s *Service) ValidateCertSecret(ctx context.Context, h *helper.Helper, name
 		h.GetClient(),
 		5*time.Second)
 	if err != nil {
-		return "", ctrlResult, err
+		return "", err
 	} else if (ctrlResult != ctrl.Result{}) {
-		return "", ctrlResult, nil
+		return "", k8s_errors.NewNotFound(
```

Comment:
this is basically the change to return a NotFound error in case VerifySecret returns a ctrlResult

Reply:
This will remove the desired requeue time though (the one coming from `VerifySecret()` and then cause exponential requeue times via the error returned?

Reply:
depends on how the caller handles the error. for those who watch the secret they reconcile when its there for the event. for others they can return a requeue time